### PR TITLE
Added dart implementation

### PR DESCRIPTION
**Pull request by SreedevT**

### CODE REVIEW

**Comment by RedRozio** [comment]
Perhaps we should use a lambda function. I.e.
dart```
bool isPrime(int n) => false;
```

File: optimized_implementations/main.dart [hunk 1]
```diff
@@ -0,0 +1 @@
+bool is_prime(int n){return false;}
```
Comment (JAicewizard) [inline optimized_implementations/main.dart:1]: Using `_` for unused parameters is more idiomatic.
Also using the `=>` syntax would avoid a lot of overhead of `return`

Comment (SreedevT) [reply optimized_implementations/main.dart]: yup incorporated changes @RedRozio

Comment (JAicewizard) [reply optimized_implementations/main.dart]: Ah I see that while I was checking the exact style guide someone was first. Although I so still sadly see one major flaw, false is one character slower than `0!=0` but I am not sure if it is worth opening another PR for such a small performance improvement

**Comment by SreedevT** [comment]
@JAicewizard nothing should stand in the way of character performance my friend.. go ahead and open that PR.

**Pull request merged**